lib/livebook_web/live/session_live/insert_buttons_component.ex

```diff
@@ -137,60 +140,84 @@ defmodule LivebookWeb.SessionLive.InsertButtonsComponent do
     """
   end
 
-  defp smart_cell_insert_button(assigns) do
+  defp code_block_insert_button(assigns) do
     ~H"""
-    <button role="menuitem" phx-click={on_smart_cell_click(@definition, 0, @section_id, @cell_id)}>
+    <button
+      role="menuitem"
+      phx-click={on_code_block_click(@definition, 0, @runtime, @section_id, @cell_id)}
+    >
+      <.remix_icon icon="terminal-box-line" />
       <span><%= @definition.name %></span>
     </button>
     """
   end
 
-  defp on_smart_cell_click(%{requirement: nil} = definition, _variant_idx, section_id, cell_id) do
-    insert_smart_cell(definition, section_id, cell_id)
+  defp smart_cell_insert_button(assigns) when is_many(assigns.definition.requirement_presets) do
+    ~H"""
+    <.submenu>
+      <:primary>
+        <button role="menuitem">
+          <span><%= @definition.name %></span>
+        </button>
+      </:primary>
+      <.menu_item :for={{preset, idx} <- Enum.with_index(@definition.requirement_presets)}>
+        <button
+          role="menuitem"
+          phx-click={on_smart_cell_click(@definition, idx, @section_id, @cell_id)}
+        >
+          <span><%= preset.name %></span>
+        </button>
+      </.menu_item>
+    </.submenu>
+    """
   end
 
-  defp on_smart_cell_click(%{requirement: %{}} = definition, variant_idx, section_id, cell_id) do
-    variant = Enum.fetch!(definition.requirement.variants, variant_idx)
+  defp smart_cell_insert_button(assigns) do
+    ~H"""
+    <button role="menuitem" phx-click={on_smart_cell_click(@definition, @section_id, @cell_id)}>
+      <span><%= @definition.name %></span>
+    </button>
+    """
+  end
 
-    with_confirm(
-      JS.push("add_smart_cell_dependencies",
-        value: %{kind: definition.kind, variant_idx: variant_idx}
```

Comment:
I moved this confirm to the server too, this way they are more aligned, and it should reduce the payload.